README.md

```diff
@@ -247,8 +257,82 @@ nginx config, deproxy response, addr and port can use templates
 in format `${part_variable}` where `part` is one of 'server',
 'tempesta', 'client' or 'backend'
 
+Warning: deproxy backend now is running on the Framework host, so
+framework ip should be specified in tempesta config.
+
 Example tests can be found in `selftests/test_framework.py`
 
 Tests can be skipped or marked as expected to fail.
 More info at [Python documentation](https://docs.python.org/3/library/unittest.html).
 
+## Internal structure and motivation of user configured tests
+
+User configured tests have very flexible structure. They allow arbitrary
+clients and server start, stop, making requests. This leads to several
+points in internal structure.
+
+### Using separate thread for polling cycle
+
+Now, deproxy client and deproxy server, all of them use the single polling
+cycle, as it was in functional tests. But we have differencies.
+
+We have 3 cases of using deproxy clients and server:
+
+1) both deproxy client and server are used
+
+2) only deproxy client is used
+
+3) only deproxy server is used
+
+First case corresponds to functional tests. The second and third have no
+corresponding case in old testing framework.
+
+And case 3 leads to instant running of polling cycle in separate thread.
+Indeed, let's consider case of wrk client and deproxy server without instant
+running of this cycle. We start deproxy server, then we start wrk client
+and after this we start polling cycle. The time before starting cycle,
+wrk will get an errors. Ok, let's start polling cycle before wrk. But now it's
+impossible to start wrk, because we are in polling cycle. This problem appeares,
+because with running polling cycle in the same thread, as the main procedure,
+deproxy server can recieve requests only after polling cycle starts.
```

Comment:
There was no such problem in old tests since `wrk` was not supposed to be used against `deproxy.server`.

'Polling cycle' is a bit unique and unclear term, I'll use `event loop` since it's well-known ant it's a basis for deproxy-based tests.

There is a better solution. How the deproxy test in the old tests worked: 
- The framework creates a bunch of clients and server instances
- To process socket events `python.asyncore` event loop is started.
- deproxy client has method `writable()` which returns `true` if client has data to send. Clients are not set to be `writable` until all connections are established.
- all socket events are processed in event loop
- event loop is stopped when test duration is expired.

OK, you NEED to run `wrk` against `deproxy.Server`. But why you cant create separate thread for `wrk` and start `wrk` only after that 

> until all connections are established.

event? 

Looks much simpler, no that hard-to-debug pooling locks existing in the new framework.